pkg/virt-controller/watch/vmi.go

```diff
@@ -380,6 +380,43 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
 		}
 
+		var networkPatchNeeded bool
+		// IF the VMI has a pod network, Check and patch if IP on VMI is not same as the one on Pod IP
+		for _, network := range vmiCopy.Spec.Networks {
+			if network.NetworkSource.Pod == nil {
+				continue
+			}
+			for i, iface := range vmiCopy.Status.Interfaces {
+				if iface.Name == network.Name {
+					vmiCopy.Status.Interfaces[i].IP = pod.Status.PodIP
+					networkPatchNeeded = true
+
+					if iface.IPs == nil {
+						continue
+					}
+					for j, ip := range iface.IPs {
+						if ip == iface.IP {
+							vmiCopy.Status.Interfaces[i].IPs[j] = pod.Status.PodIP
```

Comment:
Is this correct? For example: if user sets IP manually, after migration we change all the IPs on a given interface to the Pod's IP?

Reply:
I'm not sure why/how user would set Pod IP manually? is it possible to change that?
We're just changing the IP on Interface with Pod network to the IP of the new migrated virt-launcher pod.
@rmohr @phoracek your views?

Reply:
the podIP can't be set by users. Right now in theory users can mess up our VMI status because we don't protect the status struct yet. We have to enable at some point the status subresource support. But anyway, our controllers owns status completely and can set or override there everything.